src/atomsbase_interface.jl

```diff
@@ -0,0 +1,80 @@
+#
+# Interface between AtomsBase and GeomOpt that provides 
+# utility functions for manipulating systems.
+#
+# This interface helps defining which particles in the system are considered 
+# optimizable. Note that by default all particles are assumed to be optimizable. 
+# The user can call clamp_atoms to fix atoms whose position should not be optimized.
+#
+# IMPORTANT: Note that we always work in cartesian coordinates.
+#
+export update_positions, update_optimizable_coordinates, clamp_atoms
+
+
+@doc raw"""
+    update_postions(system::AbstractSystem, positions::Vector{<:AbstractVector{<:Real}}) where {L <: Unitful.Length}
+
+Creates a new system based on ``system`` but with atoms positions updated to the ones specified in `positions`, 
+using cartesian coordinates.
+"""
+function update_positions(system::AbstractSystem, positions::AbstractVector{<:AbstractVector{L}}) where {L <: Unitful.Length}
+    particles = [
+                 Atom(atom, position=SVector{3,L}(position)) 
+                 for (atom, position) in zip(system.particles, positions)]
```

Comment:
This is a problematic

```julia
julia> FlexibleSystem <: AbstractSystem
true

julia> FastSystem <: AbstractSystem
true
```

But `FastSystem` does not have `.particles`. AtomsBase does define iterator over particles, so you can just do

```julia
for (atom, position) in zip(system, positions)]
```
and it will work.

With this you can change

```julia
unction update_positions(system::AbstractSystem, positions::AbstractVector{<:AbstractVector{L}}) where {L <: Unitful.Length}
```

to 

```julia
unction update_positions(system, positions::AbstractVector{<:AbstractVector{L}}) where {L <: Unitful.Length}
```

Also `AbstractSystem` is actually an abstract type and calls like `AbstractSystem(system, particles=particles)` do call concrete types, in this case `FlexibleSystem`. It might be more clear to just call it directly.

Reply:
I agree with your comment,  but I think that in the long run we should have an update constructor for `AbstractSystem` that updates positions by automatically dispatches to appropriate subtype constructor. 

Currently everything is converted to `FlexibleSystem`.